lib/reader/src/parser.rs

```diff
@@ -955,6 +1052,12 @@ impl<'a> Parser<'a> {
         ctx.function.layout.append_inst(inst, ebb);
         ctx.map.def_entity(inst.into(), &opcode_loc).expect("duplicate inst references created");
 
+        if let Some(encoding) = encoding {
+            ctx.function.encodings.push(encoding);
+        } else {
+            ctx.function.encodings.push(Encoding::default());
+        }
```

Comment:
Here, the `encodings` table is a *secondary* entity map. Update it like this:

```rust
*ctx.function.encodings.ensure(inst) = encoding;
```

You don't need the default arm. That's already the default value.